rpc.go

```diff
@@ -69,19 +92,49 @@ func (cs *ThriftRpcImpl) GetValuesSingle(req *gen.SingleHFileKeyRequest) (r *gen
 		}
 		if ok {
 			found++
-			if !req.GetCountOnly() {
+			if !req.CountOnly {
 				res.Values[int32(idx)] = value
 			}
 		}
 	}
 
 	if Settings.debug {
-		log.Printf("[GetValuesSingle] %s found %d of %d.\n", *req.HfileName, found, len(req.SortedKeys))
+		log.Printf("[GetValuesSingle] %s found %d of %d.\n", req.HfileName, found, len(req.SortedKeys))
 	}
-	res.KeyCount = &found
+	res.KeyCount = found
 	return res, nil
 }
 
+func (g *GrpcImpl) GetValuesSingle(_ context.Context, req *pb.SingleHFileKeyRequest) (*pb.SingleHFileKeyResponse, error) {
```

Comment:
Glad to see the intention to start threading context objects throughout all these request chains.